From 76cc30b30fe2ffe332dacd5e2f9029731b0b439c Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 18:35:10 -0400
Subject: [ethernet] bnx2: Call dev_kfree_skby_any instead of dev_kfree_skb

Message-id: <1409942115-26825-4-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91497
O-Subject: [RHEL7.1 PATCH 3/8] bnx2: Call dev_kfree_skby_any instead of dev_kfree_skb.
Bugzilla: 1092495
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1092495

Upstream commit:
commit f458b2ee93ee3606c83f76213fbe49e026bac754
Author: Eric W. Biederman <ebiederm@xmission.com>
Date:   Tue Mar 11 14:17:41 2014 -0700

    bnx2: Call dev_kfree_skby_any instead of dev_kfree_skb.

    Replace dev_kfree_skb with dev_kfree_skb_any in functions that can
    be called in hard irq and other contexts.

    Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2.c b/drivers/net/ethernet/broadcom/bnx2.c
index 924229e..51556e8 100644
--- a/drivers/net/ethernet/broadcom/bnx2.c
+++ b/drivers/net/ethernet/broadcom/bnx2.c
@@ -2886,7 +2886,7 @@ bnx2_tx_int(struct bnx2 *bp, struct bnx2_napi *bnapi, int budget)
   sw_cons = BNX2_NEXT_TX_BD(sw_cons);
 
   tx_bytes += skb->len;
-  dev_kfree_skb(skb);
+  dev_kfree_skb_any(skb);
   tx_pkt++;
   if (tx_pkt == budget)
    break;
@@ -6633,7 +6633,7 @@ bnx2_start_xmit(struct sk_buff *skb, struct net_device *dev)
 
  mapping = dma_map_single(&bp->pdev->dev, skb->data, len, PCI_DMA_TODEVICE);
  if (dma_mapping_error(&bp->pdev->dev, mapping)) {
-  dev_kfree_skb(skb);
+  dev_kfree_skb_any(skb);
   return NETDEV_TX_OK;
  }
 
@@ -6726,7 +6726,7 @@ dma_error:
           PCI_DMA_TODEVICE);
  }
 
- dev_kfree_skb(skb);
+ dev_kfree_skb_any(skb);
  return NETDEV_TX_OK;
 }
 
-- 
1.7.1